service/lib/dinstaller/users.rb

```diff
@@ -60,22 +62,44 @@ def first_user
       [user.full_name, user.name, config.login.autologin_user == user, {}]
     end
 
-    def assign_first_user(full_name, user_name, password, auto_login)
+    def assign_first_user(full_name, user_name, password, auto_login, _data)
       # at first remove previous first user
-      config.users.reject(&:root?).map(&:id).each { |id| config.users.delete(id) }
+      old_users = config.users.reject(&:root?)
+      config.detach(old_users) unless old_users.empty?
+
       return if user_name.empty? # empty is used to remove first user
 
       user = Y2Users::User.new(user_name)
       user.gecos = [full_name]
       user.password = Y2Users::Password.create_plain(password)
-      config.attach_element(user)
+      config.attach(user)
+      config.login ||= Y2Users::LoginConfig.new
       config.login.autologin_user = auto_login ? user : nil
     end
 
+    def write(_progress)
+      without_run do
+        system_config = Y2Users::ConfigManager.instance.system(force_read: true)
+        target_config = system_config.copy
+        Y2Users::ConfigMerger.new(target_config, config).merge
+
+        writer = Y2Users::Linux::Writer.new(target_config, system_config)
+        issues = writer.write
+        logger.error(issues.inspect) unless issues.empty?
+      end
+    end
+
   private
 
     attr_reader :logger
 
+    def without_run(&block)
```

Comment:
`without_run` sounds confusing unless you know what we are talking about. What about something `without_run_mount` or something similar?